test/06-basic-text-test.py

```diff
@@ -564,34 +558,52 @@ def test_elem_match(self):
 
 
 # Test numeric strings for $text
-class NumStringTests(mango.NumStringDocsTextTests):
+@unittest.skipUnless(mango.has_text_service(), "requires text service")
+class NumStringTests(mango.DbPerClass):
 
     @classmethod
     def setUpClass(klass):
-        raise unittest.SkipTest('text index service not available')
-
-    def test_floating_point_val(self):
-        float_point_string = num_string_docs.DOCS[2]["number_string"]
-        q = {"$text": float_point_string}
-        docs = self.db.find(q)
-        assert len(docs) == 1
-        assert docs[0]["number_string"] == float_point_string
-
-    def test_hex_floating_point_val(self):
-        hex_float_point_string = num_string_docs.DOCS[3]["number_string"]
-        q = {"$text": hex_float_point_string}
-        docs = self.db.find(q)
-        assert len(docs) == 1
-        assert docs[0]["number_string"] == hex_float_point_string
+        super(NumStringTests, klass).setUpClass()
+        klass.db.recreate()
+        klass.db.create_text_index()
 
     def test_nan_val(self):
+        doc = {"number_NaN": "NaN"}
+        self.db.save_doc(doc)
         q = {"$text": "NaN"}
         docs = self.db.find(q)
-        assert len(docs) == 1
-        assert docs[0]["number_string"] == "NaN"
+        print docs
+        assert docs[0]["number_NaN"] == "NaN"
 
     def test_infinity_val(self):
+        doc = {"number_Infinity": "Infinity"}
+        self.db.save_doc(doc)
         q = {"$text": "Infinity"}
         docs = self.db.find(q)
-        assert len(docs) == 1
-        assert docs[0]["number_string"] == "Infinity"
+        assert docs[0]["number_Infinity"] == "Infinity"
+
+    @given(float_point_string=st.floats().map(str))
+    def test_floating_point_val(self,float_point_string):
+        assume(float_point_string!="nan")
```

Comment:
You can actually make the following strategy: `st.floats.filter(math.isfinite).map(str)` since by this `assume` you filter NaN's, but not infs.  Or that was your idea?